lib/linux_admin/hosts.rb

```diff
@@ -36,6 +38,10 @@ def update_entry(address, hostname, comment = nil)
       end
     end
 
+    def hostname=(name)
+      run!(cmd('hostnamectl'), :params => ['set-hostname', name])
```

Comment:
So traditionally `/etc/hostname` and the `hostname` command were used to deal with this.

With systemd systems we got the new `systemd-hostnamed` service which seems to be adding a layer of complexity to this.

For example:
I would have expected running `hostnamectl set-hostname new-host` to have the same effect as `echo "new-host" > /etc/hostname && hostname -F /etc/hostname`, but after running the latter commands `hostnamectl status` still shows the old name and doesn't change until I run `systemctl restart systemd-hostnamed`

So I'm thinking like this:

```ruby
if cmd?("hostnamectl")
  run!(cmd('hostnamectl'), :params => ['set-hostname', name])
else
  File.write("/etc/hostname", name)
  run!(cmd('hostname'), :params => ["-F", "/etc/hostname"])
end
```
What do you think @Fryguy ?